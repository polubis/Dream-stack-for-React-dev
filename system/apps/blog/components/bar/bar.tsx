import { ArrowTopIcon, Button, Bar as UIBar } from '@system/figa-ui';

import type { BarProps } from './defs';
import { useScrollTo } from '@system/figa-hooks';

const Bar = ({ children, ...props }: BarProps) => {
  const [, { toTop }] = useScrollTo();

  return (
    <UIBar right {...props}>
      {children}
      <Button shape="rounded" size={2} onClick={() => toTop()}>
        <ArrowTopIcon />
      </Button>
    </UIBar>
  );
};

export { Bar };
