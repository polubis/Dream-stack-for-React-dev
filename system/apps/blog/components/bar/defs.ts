import { BarProps as UIBarProps } from '@system/figa-ui';

type BarProps = Partial<UIBarProps>;

export type { BarProps };
