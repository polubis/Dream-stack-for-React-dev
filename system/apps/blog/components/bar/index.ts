export * from './bar';
export * from './defs';
