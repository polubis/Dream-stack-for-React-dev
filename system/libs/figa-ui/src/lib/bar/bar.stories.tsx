import type { Story, Meta } from '@storybook/react';

import { Bar } from './Bar';
import type { BarProps } from './defs';
import { Button } from '../button';
import { ArrowTopIcon, CloseIcon } from '../icon';

export default {
  component: Bar,
  title: 'Bar',
} as Meta;

const Nodes = [
  <Button>
    <ArrowTopIcon />
  </Button>,
  <Button>
    <CloseIcon />
  </Button>,
];

const Template: Story<BarProps> = () => {
  return (
    <>
      <Bar>{Nodes}</Bar>
      <Bar right>{Nodes}</Bar>
      <Bar top right>
        {Nodes}
      </Bar>
      <Bar top>{Nodes}</Bar>
    </>
  );
};

export const Default = Template.bind({});
Default.args = {};
