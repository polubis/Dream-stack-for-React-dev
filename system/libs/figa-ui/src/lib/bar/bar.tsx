import styled from 'styled-components';
import type { BarProps } from './defs';
import c from 'classnames';
import { column } from '../shared';
import { tokens } from '../theme-provider';

const Container = styled.div`
  ${column()}
  position: fixed;
  bottom: ${tokens.spacing[250]};
  left: ${tokens.spacing[250]};
  transition: 0.3s transform ease-in-out;
  transform: scale(1);

  & > *:not(:last-child) {
    margin-bottom: ${tokens.spacing[150]};
  }

  &.hidden {
    transform: scale(0);
  }

  &.right {
    left: unset;
    right: ${tokens.spacing[250]};
  }

  &.top {
    bottom: unset;
    top: ${tokens.spacing[250]};
  }
`;

const Bar = ({ className, children, top, right, hidden }: BarProps) => {
  return (
    <Container className={c('bar', className, { top }, { right }, { hidden })}>
      {children}
    </Container>
  );
};

export { Bar };
