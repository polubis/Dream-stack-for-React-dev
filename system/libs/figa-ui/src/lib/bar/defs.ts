import type { ReactNode } from 'react';

interface BarProps {
  className?: string;
  children: ReactNode;
  top?: boolean;
  right?: boolean;
  hidden?: boolean;
}

export type { BarProps };
