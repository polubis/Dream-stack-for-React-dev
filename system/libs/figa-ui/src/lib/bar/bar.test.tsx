import { render } from '@testing-library/react';
import { Bar } from './bar';

describe('Bar can be used when: ', () => {
  it('[FRAGILE] assigns default classes', () => {
    const { asFragment } = render(
      <Bar className="my-class">
        <button>A</button>
        <button>B</button>
      </Bar>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] assigns given positions', () => {
    const { asFragment } = render(
      <Bar top right>
        <button>A</button>
      </Bar>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] can be hidden', () => {
    const { asFragment } = render(
      <Bar hidden>
        <button>A</button>
      </Bar>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
