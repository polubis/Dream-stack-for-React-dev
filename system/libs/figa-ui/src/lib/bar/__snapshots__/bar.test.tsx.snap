// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar can be used when:  [FRAGILE] assigns default classes 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe AVuto bar my-class"
  >
    <button>
      A
    </button>
    <button>
      B
    </button>
  </div>
</DocumentFragment>
`;

exports[`Bar can be used when:  [FRAGILE] assigns given positions 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe AVuto bar top right"
  >
    <button>
      A
    </button>
  </div>
</DocumentFragment>
`;

exports[`Bar can be used when:  [FRAGILE] can be hidden 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe AVuto bar hidden"
  >
    <button>
      A
    </button>
  </div>
</DocumentFragment>
`;
